index.js

```diff
@@ -54,6 +59,17 @@ module.exports = function * (next) {
     }
   }
 
+  //Adjust infinite end
+  if (end == Number.POSITIVE_INFINITY) {
+    if(isNaN(len)) {
```

Comment:
@ZetaTwo anyway could you please fix the above nits, and we will have a review again, thanks :-)